lm75bd/lm75bd.c

```diff
@@ -27,6 +28,28 @@ error_code_t lm75bdInit(lm75bd_config_t *config) {
 
 error_code_t readTempLM75BD(uint8_t devAddr, float *temp) {
   /* Implement this driver function */
+
+  error_code_t errCode;
+
+  if (temp == NULL) {
+    return ERR_CODE_INVALID_ARG;
+  }
+
+  uint8_t tempPointerByte = 0x00U; // selects temp register at 00 LSB
+
+  uint8_t *sendBuf = &tempPointerByte;
+  RETURN_IF_ERROR_CODE(i2cSendTo(devAddr, sendBuf, sizeof(sendBuf)));
+
+  uint8_t receiveBuf[2] = {0};
+  RETURN_IF_ERROR_CODE(i2cReceiveFrom(devAddr, receiveBuf, sizeof(receiveBuf)));
+
+  uint16_t MSByte = (uint16_t) receiveBuf[0];
+  uint16_t LSByte = (uint16_t) receiveBuf[1];
```

Comment:
It might be because you are using uint16_t here and the bit shifting causes issues. Use uint8_t instead as I have seen this work for it before